contrib/core-contract-tests/tests/pox-4/pox_RevokeDelegateStxCommand.ts

```diff
@@ -0,0 +1,96 @@
+import { PoxCommand, Real, Stub, Wallet } from "./pox_CommandModel.ts";
+import { poxAddressToTuple } from "@stacks/stacking";
+import { expect } from "vitest";
+import { Cl, someCV, tupleCV } from "@stacks/transactions";
+
+/**
+ * The `RevokeDelegateStxCommand` revokes the delegation for stacking within PoX-4.
+ *
+ * Constraints for running this command include:
+ * - The `Stacker` has to currently be delegating.
+ */
+export class RevokeDelegateStxCommand implements PoxCommand {
+  readonly wallet: Wallet;
+
+  /**
+   * Constructs a RevokeDelegateStxCommand to revoke delegate uSTX for stacking.
+   *
+   * @param wallet - Represents the Stacker's wallet.
+   */
+  constructor(wallet: Wallet) {
+    this.wallet = wallet;
+  }
+
+  check(model: Readonly<Stub>): boolean {
+    // Constraints for running this command include:
+    // - The Stacker has to currently be delegating.
+    return (
+      model.stackingMinimum > 0 &&
+      model.wallets.get(this.wallet.stxAddress)!.hasDelegated === true
+    );
+  }
+
+  run(model: Stub, real: Real): void {
+    // Get the Operator's wallet
+    const operatorWallet = model.wallets.get(this.wallet.delegatedTo)!;
+
+    // Act
+    const revokeDelegateStx = real.network.callPublicFn(
+      "ST000000000000000000002AMW42H.pox-4",
+      "revoke-delegate-stx",
+      [],
+      this.wallet.stxAddress,
+    );
+
+    // Assert
+    expect(revokeDelegateStx.result).toBeOk(
+      someCV(
+        tupleCV({
+          "amount-ustx": Cl.uint(this.wallet.delegatedMaxAmount),
+          "delegated-to": Cl.principal(
+            operatorWallet.stxAddress || "",
+          ),
+          "pox-addr": Cl.some(
+            poxAddressToTuple(operatorWallet.btcAddress || ""),
```

Comment:
I believe this would only pass if the PoX address previously belonged to the operator? Might make sense to fetch the existing pox-address first? I don't think it necessarily has to be the PoX address derived from the operator key?

Reply:
Yes, it depends on how the delegation is performed. Now, we pick a random address and delegate to it (using it's `btcAddress`, too).